### ISSUE TITLE
Clarification on refined batch sampling

### ISSUE THREAD

**Issue by LeeHoang2710** [jaeseokbyun/RTD#1]
Hi authors,

Thank you for releasing the code and paper — I’ve been studying your Refined Batch Sampling Strategy for Hard Negatives and noticed a small but potentially important mismatch between the paper and the released code.

In Section “Refined batch sampling strategy for hard negatives”, the paper states that each batch incorporates pairs of (T₍r+c₎, Tₜ) and (Tᵣ, Tᵣ).
For example:

A pair such as (T₍r+c₎: “A dog catching a frisbee + change dog to cat”, Tₜ: “A cat catching a frisbee”) is sampled along with its corresponding reference pair (Tᵣ: “A dog catching a frisbee”, Tᵣ: “A dog catching a frisbee”) in the same batch.

In the released training code, however, I found the following:

#### Use refined batch sampling
concat_tokens = torch.cat((concat_tokens, special_tokens), dim=0).to(accelerator.device)
target_tokens = torch.cat((target_tokens, reference_tokens), dim=0).to(accelerator.device)

This means that the second half of the batch pairs special_tokens (“a photo of [$]”) with reference_tokens, rather than the expected (Tᵣ, Tᵣ) pair.

Could you clarify why the code uses special_tokens (“a photo of [$]”) instead of a second copy of reference_tokens for the refined batch sampling step?

Thank you.

**Comment by jaeseokbyun** [comment]
Thank you for your patience, and apologies for the delayed response and any confusion.

This is related to the refined concatenation scheme, where we use the template "a photo of [$] that [T_c]" with [$] replaced by output of phi (noise-injected). Since we apply this technique to the first part of (T₍r+c₎, Tᵣ) pair, we adopt a similar approach for the (Tᵣ, Tᵣ) pair when encoding its first component.

In the paper, we introduced the refined batch sampling strategy first, followed by the refined concatenation scheme for readability. So that we omitted this detail for readibility. We will clarify this further in the code comments to avoid confusion.

Apologies for any confusion this may have caused.
